File: src/main/java/ru/tyom/test_bot/bot/SimpleBot.java
package ru.tyom.test_bot.bot;

import org.telegram.telegrambots.longpolling.util.LongPollingSingleThreadUpdateConsumer;
import org.telegram.telegrambots.meta.api.objects.Update;
import org.telegram.telegrambots.meta.generics.TelegramClient;
import ru.tyom.test_bot.config.TelegramBotProperties;
import ru.tyom.test_bot.service.MessageHandler;

public class SimpleBot implements LongPollingSingleThreadUpdateConsumer {

	private final MessageHandler messageHandler;
	private final TelegramClient telegramClient;
	private final String botUsername;

	public SimpleBot(TelegramBotProperties properties, MessageHandler messageHandler, TelegramClient telegramClient) {
		this.messageHandler = messageHandler;
		this.telegramClient = telegramClient;
		this.botUsername = properties.getUsername();
	}

	@Override
	public void consume(Update update) {
		messageHandler.handle(update, telegramClient);
	}

	public String getBotUsername() {
		return botUsername;
	}
}

File: src/main/kotlin/ru/tyom/test_bot/bot/SimpleBot.kt
package ru.tyom.test_bot.bot

import org.telegram.telegrambots.longpolling.util.LongPollingSingleThreadUpdateConsumer
import org.telegram.telegrambots.meta.api.objects.Update
import org.telegram.telegrambots.meta.generics.TelegramClient
import ru.tyom.test_bot.config.TelegramBotProperties
import ru.tyom.test_bot.service.MessageHandler

class SimpleBot(
    properties: TelegramBotProperties,
    private val messageHandler: MessageHandler,
    private val telegramClient: TelegramClient
) : LongPollingSingleThreadUpdateConsumer {

    private val botUsername: String = properties.username ?: ""

    override fun consume(update: Update) {
        messageHandler.handle(update, telegramClient)
    }

    fun getBotUsername(): String = botUsername
}
